lib/actions/field-trip.js

```diff
@@ -249,17 +406,355 @@ function planInbound (request) {
     }, config)
     // clearTrip()
     const queryParams = {
-      date: moment(request.travelDate).format(OTP_API_DATE_FORMAT),
+      date: getFormattedRequestTravelDate(request),
       departArrive: 'DEPART',
-      groupSize: getGroupSize(request),
       time: moment(request.leaveDestinationTime).format(OTP_API_TIME_FORMAT),
       ...locations
     }
     dispatch(setQueryParam(queryParams))
-    dispatch(routingQuery())
+    dispatch(makeFieldTripPlanRequests(request))
+  }
+}
+
+/**
+ * Makes appropriate OTP requests until enough itineraries have been found to
+ * accommodate the field trip group.
+ */
+function makeFieldTripPlanRequests (request) {
+  return async function (dispatch, getState) {
+    const prePlanState = getState()
+    const fieldTripModuleConfig = getModuleConfig(
+      prePlanState,
+      Modules.FIELD_TRIP
+    )
+
+    // request other known trip IDs that other field trips are using on the
+    // field trip request date
+    try {
+      await getTripIdsForTravelDate(dispatch, prePlanState, request)
+    } catch (e) {
+      alert(
+        `Error fetching trips for field trip travel date: ${JSON.stringify(e)}`
+      )
+      return
+    }
+
+    // create a new searchId to use for making all requests
+    const searchId = randId()
+
+    // set numItineraries param for field trip requests
+    dispatch(setQueryParam({ numItineraries: 1 }))
+
+    // create a new set to keep track of trips that should be banned
+    const bannedTrips = new Set()
+
+    // track number of requests made such that endless requesting doesn't occur
+    const maxRequests = fieldTripModuleConfig?.maxRequests || 10
```

Comment:
Perhaps we should just return a default field trip config from `getModuleConfig` if it does not exist, so that we can safely destructure (and avoid the optional chaining)?

Reply:
I'd prefer the `getModuleConfig` method to return something non-truthy if a module doesn't exist.